Resources/public/js/views/services/ez-contenteditviewservice.js

```diff
@@ -66,6 +69,85 @@ YUI.add('ez-contenteditviewservice', function (Y) {
         },
 
         /**
+         * Sets fields of edited version.
+         *
+         * @method _setVersionFields
+         * @private
+         * @param {Function} callback
+         */
+        _setVersionFields: function (callback) {
+            var service = this,
+                content = this.get('content'),
+                resources = content.get('resources'),
+                contentType = resources.ContentType,
+                version = this.get('version'),
+                fields = content.get('fields'),
+                languageCode = this.get('languageCode'),
+                baseLanguageCode = this.get('baseLanguageCode'),
+                languageCodeForLoadContent = baseLanguageCode ? baseLanguageCode : languageCode,
+                setDefaultFields = false,
+                languageInTranslationsList;
+
+            languageInTranslationsList = Y.Array.find(
+                                            content.get('currentVersion').getTranslationsList(), 
+                                            function (translation) {
+                                                if (translation === languageCodeForLoadContent) {
+                                                    return true;
+                                                }
+                                            }
+                                        );
+
+            if (baseLanguageCode) {
+                if (languageInTranslationsList === null && !content.get('alwaysAvailable')) {
+                    this._error(
+                        "Could not load the content with id '" + content.get('contentId')
+                        + "' and languageCode '" + languageCodeForLoadContent + "'"
+                    );
+                    return;
+                }
+            } else {
+                if ( languageInTranslationsList === null) {
+                    setDefaultFields = true;
+                }
+            }
+
+            this._loadContentType(contentType, function () {
+                if (setDefaultFields) {
+                    fields = service._getDefaultFields(service.get('contentType'), service.get('languageCode'));
+                } else {
+                    Y.each(fields, function (field) {
```

Comment:
you are modifying the fields of the content, you should create a new `fields` object containing clones of the field on which you set the language code.